src/dune_rules/menhir/menhir_rules.ml

```diff
@@ -263,26 +265,32 @@ module Run (P : PARAMS) = struct
      either [process3] or [process1], as appropriate. *)
 
   (* Because Menhir processes [--only-tokens] before the [--infer-*] commands,
-     when [--only-tokens] is present, no [--infer-*] command should be used. *)
+     when [--only-tokens] is present, no [--infer-*] command should be used.
+     When Menhir is called with [--only-tokens], it does not generate a [.conflicts]
+     file even the [--explain] flag is passed. Otherwise, a (possibly empty)
+     [.conflicts] file is always generated (if Menhir succeeds in creating a
+     [.ml/.mli] file).
+  *)
 
   let process (stanza : stanza) : unit Memo.t =
     let base = Option.value_exn stanza.merge_into in
-    let ocaml_type_inference_disabled, cmly =
+    let ocaml_type_inference_disabled, cmly, conflicts =
       Ordered_set_lang.Unexpanded.fold_strings
         stanza.flags
```

Comment:
Upon reading it seems that `stanza.flags` doesn't contain `--explain` yet if it comes from the env. --explain from the env only becomes available in process3 it seems. It's tricky to know --explain is in env without first expanding and I got stuck with my PR because it is too early to expand here.

Reply:
I'm not sure I get the subtlety, but here is what I could test:
- dune does not seem to recognize .conflicts as targets (is it because they are `Hidden_targets` ?) 
```
Error: Don't know how to build _build/default/a/parser.conflicts
```
- it does not seem to be related to `env`. The other test I added (which uses `flags` in the menhir stanza directly) has the same problem (.conflicts not being a target)
- This is also the case for cmly files (Don't know how to build).
- This is also the case for .s files produced by `ocamlopt`. Indeed, if I put `(ocamlopt_flags -S)`, the .s assembly files are present in _build/.../native/foo__bar.s but I have the same error when I try to `dune build` them.

In any case, I don't think this is an issue for most menhir users (or at least no more than the other two cases that don't "work" either and are unrelated to this flag). 
These .conflicts files are really transient files that are used to debug one's grammar, they are not to be committed to a projects repository (at least no more than .s files generated by ocamlopt). One exception would be `menhir` developers that would have a use for that, but this is another, far less common use case.

Reply:
Right, but you get the same behaviour with menhir today if you pass `--explain` before your PR. What your PR tries to do is declare `.conflict` files as targets of the menhir rule when `--explain` is passed in the stanza. Being a hidden target doesn't matter here, Dune should still be able to build it.

ocamlopt_flags -S is also not a good example because Dune does not know anything about -S and so does not produce the .s files as targets of the OCaml module rules. This means that those files that are produced will not be known by Dune hence they can be deleted or cleaned up causing users to miss them.

I definitely agree that this is a useful feature to have, but it has proven to be tricky to implement correctly. In my old PR #7607 you should see that `.conflicts` files are at least recognized when `--explain` is passed in the stanza. For env the situation is a bit trickier, but I would be OK with having a partial solution where `--explain` works in a stanza but not in env.

Reply:
I don't get the same behavior. I have a more complex project where the .conflicts files are not present in _build. My understanding is that it is due to sandboxing but when I use --sandbox=none the file still isn't there (and I still see in the logs, cd .sandbox/... before invocations of menhir).

If I build my project with this PR, the .conflicts files appears (regardless of --sandbox). Or is it possible I picked up a commit between 3.11.1 and main that fixes this ?

Reply:
Arrrg. Scracth that. I understand now what you mean (sorry).
So both my test cases were buggy because they use `(land dune 3.0)` but sandboxing is forced starting with `(lang dune 3.5)` for menhir. And indeed if I change the dune version, the simple test passes but not the env. (Which is still an improvement I would say).
I'll investigate further.

Reply:
@Tchou no worries. Let me know if you have any questions about the code.